accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,126 @@
+
+A developer has confidence that their library using source generators will continue to work in the same way a library without source generators works on future versions of .NET following the same compatibility rules.
+
+## Requirements
+
+### Goals
+
+- Enable consumption of source generators for components in shared frameworks
+- Address conflict resolution between source generators in shared frameworks and nuget packages
+- Define servicing process for shared framework source generators
+- Establish guidelines building and versioning shared framework source generators
+- Describe compatibility requirements for source generators.
+
+### Non-Goals
+
+- Define a process for components not in shared frameworks to contribute source generators.
+- Define any changes to the source generator architecture.
+- Define any implementation details of source generators or source generator tests.
+- Define a process around internal or non-shipping source generators
+
+## Stakeholders and Reviewers
+
+- ASP.NET - also building shared framework source generators.
+- .NET Libraries team - building multiple shared framework source generators.
+- Mono team - interested in providing source generators.
+- Roslyn team - responsible for source generator infrastructure.
+- .NET SDK team - responsible for resolving shared frameworks and doing conflict resolution with content.
+- NuGet team - FYI only, no changes requested
+
+## Design
+
+### Framework provided source generators
+
+Framework provided source generators should have a mechanism to be exposed and versioned along side the framework and library which they extend.  These source generators will be passed to the compiler when that framework is targeted.  These source generators will not ne used when the framework is not targeted.  For example, ASP.NET specific source generators will not be used with ASP.NET is not referenced.  The version of the analyzers specific to the TargetFramework version will be used.  For example, when targeting .NET 6.0 the analyzers specific to net6.0 will be used and not those for .NET 7.0, even when using the SDK that ships with .NET 7.0.
+
+To facilitate this, we will package analyzers inside [reference packs](https://github.com/dotnet/designs/blob/main/accepted/2019/targeting-packs-and-runtime-packs.md).  We will use the standard [nuget conventions for describing analyzers](https://github.com/dotnet/designs/blob/main/accepted/2019/targeting-packs-and-runtime-packs.md).  The .NET SDK will be responsible for locating the appropriate analyzers for the TargetFramework and creating `@(Analyzer)` items as part of the `ResolveTargetingPackAssets` task.  These items will be conflict-resolved with other Analyzer sources based as described in the following section.
```

Comment:
It is very rare for reference packs to need servicing today. Source generators are much more likely to contain bugs that require servicing. Should it be mentioned here? Are there any challenges that this is going to create?

Reply:
I cover this below in Q/A.  IMHO we ship *a* ref pack regardless, similar to how we ship the runtime regardless, so no matter what we are shipping the same number of bits.  We also need to make sure the ref-pack is serviceable in general.  We shouldn't let a potential bug in something that needs to work lead us down a different path.

If it weren't the ref-pack then it would be something else, in which case we'd have even more moving pieces.